plugins/modem-manager/fu-mm-device.c

```diff
@@ -1408,24 +1455,7 @@ fu_mm_device_write_firmware_firehose(FuDevice *device,
 	archive = fu_archive_new(fw, FU_ARCHIVE_FLAG_IGNORE_PATH, error);
 	if (archive == NULL)
 		return FALSE;
-	fu_progress_step_done(progress);
-
-	/* modify firmware search path and restore it before function returns */
-	locker = fu_device_locker_new_full(self,
-					   (FuDeviceLockerFunc)fu_mm_prepare_firmware_search_path,
-					   (FuDeviceLockerFunc)fu_mm_restore_firmware_search_path,
-					   error);
-	if (locker == NULL)
-		return FALSE;
-
-	/* firehose modems that switch to the EDL mode over QCDM port require
-	 * firehose binary to be present in the firmware-loader search path. */
-	if (!fu_mm_copy_firehose_prog(self, archive, error))
-		return FALSE;
-
-	/* switch to embedded downloader (EDL) execution environment */
-	if (!fu_mm_device_qcdm_switch_to_edl(FU_DEVICE(self), error))
-		return FALSE;
+	g_object_ref(archive);
```

Comment:
why ref the archive here? Something fishy is going on with the refcounting.

Reply:
Are you using the GBytes from the archive without taking a ref to those perhaps?

Reply:
aa, thanks for the tip. i didn't need to autofree `firehose_prog`, the warning is gone now